src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// DiagnosticHandler notifies DiagnosticSource subscribers about outgoing Http requests
+    /// </summary>
+    internal sealed class DiagnosticsHandler : DelegatingHandler
+    {
+        /// <summary>
+        /// DiagnosticHandler constructor
+        /// </summary>
+        /// <param name="innerHandler">Inner handler: Windows or Unix implementation of HttpMessageHandler. 
+        /// Note that DiagnosticHandler is the latest in the pipeline </param>
+        public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
+        {
+        }
+
+        internal static bool IsEnabled()
+        {
+            return s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.RequestWriteName);
+        }
+
+        protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            //HttpClientHandler is responsible to call DelegatingHandler.IsEnabled() before forwarding request to it.
+            //so this code will never be reached if RequestWriteName is not enabled
+            Guid loggingRequestId = LogHttpRequest(request);
+            Task<HttpResponseMessage> responseTask = base.SendAsync(request, cancellationToken);
+
+            if (s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.ResponseWriteName))
```

Comment:
There are actually two different strings being comared.  The first with a 'Request' suffix and the latter with a 'Response' suffix.    Since we are about to write the Response we do the check that that event is enabled.    It is the expected cliché (always surround the write with an IsEnabled check).  

One could argue that the ability to turn off 'Response' selectively even if 'Request' may not be valuable.   But since we are not in the most critical path (we have already committed to doing some logging).

In thinking about it however, I think it is better to change it so that the 'master' switch is the Response rather than the Request.   (and put the if check in the request part).   That is because if you want to be lean, you only have one event you want it to be the response (since it has the most information, and in particular the duration).    Then you can decide if you want both Request as well as the Response.